bokehjs/src/lib/models/tools/edit/poly_edit_tool.ts

```diff
@@ -56,30 +57,48 @@ export class PolyEditToolView extends PolyToolView {
     if (!this.model.active)
       return
 
+    const vsync_renderer = this.model.renderers[0]
+    const vsync_updater = () => this._update_vertices(vsync_renderer)
+    let vsync_ds = null
+    if (vsync_renderer !== undefined) vsync_ds = vsync_renderer.data_source
+
     const renderers = this._select_event(ev, "replace", this.model.renderers)
     if (!renderers.length) {
       this._set_vertices([], [])
       this._selected_renderer = null
       this._drawing = false
+      this._cur_index = null
+      if (vsync_ds !== null)
```

Comment:
```suggestion
      if (vsync_ds != null)
```

Reply:
In general one should prefer `!=` over `!==`, unless there is an explicit need for distinguishing between `undefined` and `null`, which should be utilized as rarely as possible.